File: scripts/gerar_senha_hash.py
#!/usr/bin/env python3
"""
gerar_senha_hash.py
Gera o hash SHA-256 de uma senha para usar no data/config.json

Uso:
    python3 scripts/gerar_senha_hash.py <sua-senha>

Exemplo:
    python3 scripts/gerar_senha_hash.py minhaSenha123
"""
import hashlib
import sys


def main():
    if len(sys.argv) < 2:
        print("\n❌  Uso: python3 scripts/gerar_senha_hash.py <sua-senha>\n")
        sys.exit(1)

    senha = sys.argv[1]
    hash_hex = hashlib.sha256(senha.encode("utf-8")).hexdigest()

    sep = "─" * 53
    print(f"\n{sep}")
    print("  Carômetro Escolar — Gerador de Hash de Senha")
    print(sep)
    print(f"  Senha informada : {senha}")
    print(f"  Hash SHA-256    : {hash_hex}")
    print(sep)
    print("\n  Cole o hash acima no campo \"senha_hash\" do usuário")
    print("  em data/config.json\n")


if __name__ == "__main__":
    main()

File: scripts/gerar-senha-hash.js
#!/usr/bin/env node
/**
 * gerar-senha-hash.js
 * Gera o hash SHA-256 de uma senha para usar no data/config.json
 *
 * Uso:
 *   node scripts/gerar-senha-hash.js <sua-senha>
 *
 * Exemplo:
 *   node scripts/gerar-senha-hash.js minhaSenha123
 *
 * O hash gerado deve ser colado no campo "senha_hash" do usuário
 * no arquivo data/config.json
 */

const { createHash } = require('crypto');

const senha = process.argv[2];

if (!senha) {
  console.error('\n❌  Uso: node scripts/gerar-senha-hash.js <sua-senha>\n');
  process.exit(1);
}

const hash = createHash('sha256').update(senha).digest('hex');

console.log('\n─────────────────────────────────────────────────────');
console.log('  Carômetro Escolar — Gerador de Hash de Senha');
console.log('─────────────────────────────────────────────────────');
console.log(`  Senha informada : ${senha}`);
console.log(`  Hash SHA-256    : ${hash}`);
console.log('─────────────────────────────────────────────────────');
console.log('\n  Cole o hash acima no campo "senha_hash" do usuário');
console.log('  em data/config.json\n');
